Robustness task

rimashahbazyan opened the pull request.

baberabb committed:
[Bugfix] add temperature=0 to logprobs and seed args to API models (#2149)

* add temperature for log probs

* add seed

* nit

* add new args to test

* added warning for api chat models

nathan-weinberg committed:
refactor: limit usage of `scipy` and `skilearn` dependencies (#2097)

* refactor: move scipy and sklearn module imports to func imports

Signed-off-by: Nathan Weinberg <nweinber@redhat.com>

* refactor: consolidate weighted_f1_score func into lm_eval utils

Signed-off-by: Nathan Weinberg <nweinber@redhat.com>

* lint: allow for utils file to have unused imports

this allows for shared functions to be defined only
once while allowing for the YAML function importing
to continue working

Signed-off-by: Nathan Weinberg <nweinber@redhat.com>

---------

Signed-off-by: Nathan Weinberg <nweinber@redhat.com>

haileyschoelkopf committed:
Update lm-eval-overview.ipynb (#2118)

kargaranamir committed:
fix typo. (#2169)

zhabuye committed:
Update README.md (#2125)

NathanHB committed:
Dp and mp support (#2056)

* batch commit

* :Revert "batch commit"

This reverts commit d859d1ca3bc2d6e45979ec51379df119f2a9633d.

* batch commit

* checkout from main

* checkout from main

* checkout from main

* checkout from main

* checkout from main

* cleanup

* cleanup

* cleanup

* cleanup

* cleanup

* cleanup

* cleanup

* cleanup

* linting

* add doc

* Update lm_eval/models/huggingface.py

Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

* Update README.md

Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

* Update lm_eval/models/huggingface.py

* linter

* Apply suggestions from code review

Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

* style

* remove prepare

* fix

* style

* last check

* Update lm_eval/models/huggingface.py

Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

---------

Co-authored-by: Clémentine Fourrier <22726840+clefourrier@users.noreply.github.com>
Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>
Co-authored-by: clementine@huggingface.co <clementine@huggingface.co>

baberabb committed:
[hotfix] API: messages were created twice (#2174)

kargaranamir committed:
add okapi machine translated notice. (#2168)

haileyschoelkopf committed:
remove incorrectly inherited group names (#2181)

ysjprojects committed:
Mmlu Pro (#1961)

* initialized mmlu_pro task

* added generative mmlu-pro

* added cot fewshot for mmlu-pro

* Initial commit

* updated mmlu-pro to take on 3 splits: test, val, dev

* mmlu-pro: added continuation and flan_cot_zeroshot

* added README.md for mmlu_pro

* removed

* update files

* moved files out, and removed unused versions

* updated

* mmlu_pro:

-changed task 'other' to 'miscellaneous'
there is already a group named 'other'
task and group with the same alias (e.g. mmlu_pro_other_generative) throws an error

-fixed yaml backslash escape for fewshot cot

* changed choices -> options in yaml config to fit dataset schema

* ONLY FOR DEFAULT: fixed yaml file to use variable number of choices

* mmlu-pro: fixed doc_to_text/choice/target configs for all variants

* mmlu-pro: minor fixes

* mmlu-pro/default: aligned with mmlu updates

* mmlu-pro: update yaml content in line with mmlu

* mmlu-pro: fixed mislabelling of task (math->chemistry)

* mmlu-pro: fixed yaml formatting

* add custom fewshot doc_to_text, target, and choice

* add process for each subtask

* add process for each subtask

* pre-commit

* pre-commit

* format

* resolved left out merge

* deleted folders + updated readme

* Update evaluator.py

* Update evaluator.py

---------

Co-authored-by: Yu Shi Jie <shijie@tensorplex.ai>
Co-authored-by: lintangsutawika <lintang@eleuther.ai>
Co-authored-by: root <root@455bdd73-01.cloud.together.ai>
Co-authored-by: Lintang Sutawika <lintang@sutawika.com>

ysjprojects committed:
added gsm_plus (#2103)

* added gsm_plus

* formatted dataset to have train-test-splits

* README.md for gsm-plus

* Update README.md

* GSM-Plus: added gsm_plus_mini

* GSM-Plus: attribution to original dataset

* Update README.md

* Update README.md

* Update README.md

---------

Co-authored-by: Lintang Sutawika <lintang@eleuther.ai>

haileyschoelkopf committed:
fix revision type (#2184)

haileyschoelkopf committed:
Update README.md (#2186)

ysjprojects committed:
gsm_plus minor fix (#2191)

* fixed gsm

* GSM-Plus: remove dataset_name line

jungwhank committed:
keep new line for task description (#2116)

* add keep trailing newline

* apply ruff-format

* add prompt unit test

* increment the version of tasks that have description with whitespace

* remove white spaces of leaderboard bbh

* update MMLU expected versions in output

* CI run does display the expected version=1 for mmlu subtasks, fix expected test output again

---------

Co-authored-by: haileyschoelkopf <hailey@eleuther.ai>

ysjprojects committed:
Update README.md (#2206)

antonpolishko committed:
Update citation in README.md (#2083)

Bumped citation to the v0.4.3

am-bean committed:
New task: Lingoly (#2198)

* Setting up lingoly task

* Testing yaml changes to debug

* Adding pre-commit hooks

* Functional LingOly benchmark

* Renaming files and adding grouping

* Extending group aggregations to allow custom functions. Setting up custom lingoly aggregation using difference in scores.

Cameron7195 committed:
Created a new task for gsm8k which corresponds to the Llama cot settings… (#2215)

* Created a new task for gsm8k which corresponds to the cot settings and prompt formatting described by Meta to evaluate Llama. Useful for replicating Llama performance on GSM8K benchmark.

* fixing formatting

* fixing formatting

am-bean committed:
Lingoly README update (#2228)

* Setting up lingoly task

* Testing yaml changes to debug

* Adding pre-commit hooks

* Functional LingOly benchmark

* Renaming files and adding grouping

* Extending group aggregations to allow custom functions. Setting up custom lingoly aggregation using difference in scores.

* Adding LingOly to the README file

Uminosachi committed:
Update yaml to adapt to belebele dataset changes (#2216)

adamlin120 committed:
Add TMLU Benchmark Dataset (#2093)

* add taiwan truthful qa

* add tmlu

* Add .gitignore entries for evals/ and harness_eval_main_log.txt, and add harness_eval.slurm script

* add pega eval and legal eval

* add ccp eval

* Update .gitignore and harness_eval.slurm

* Add trust_remote_code and wandb_args to harness_eval.slurm, and add run_all.sh script

* Add Pega MMLU task and configuration files

* Add new models and update parameters in run_all.sh

* Add UMTCEval tasks and configurations

* Update dataset paths and output path

* Update .gitignore and harness_eval.slurm, and modify _generate_configs.py

* Update SLURM script and add new models

* clean for pr

* Update lm_eval/tasks/tmlu/default/tmlu.yaml

Co-authored-by: Lintang Sutawika <lintang@sutawika.com>

* adjust tag name

* removed group alias from tasks

* format

---------

Co-authored-by: Lintang Sutawika <lintang@sutawika.com>
Co-authored-by: lintangsutawika <lintang@eleuther.ai>
Co-authored-by: Yen-Ting Adam, Lin <r08944064@csie.ntu.edu.tw>

lewtun committed:
Update IFEval dataset to official one (#2218)

* Update IFEval dataset to official one

This PR updates the IFEval dataset to the one hosted under the Google org: https://huggingface.co/datasets/google/IFEval

Note the main change is an updated prompt from this commit in the GitHub repo: https://github.com/google-research/google-research/commit/26d8ccdab6fec61b5c83ad6327ea8bda9e580288

* Update ifeval.yaml

---------

Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

NathanHB committed:
fix the leaderboard doc to reflect the tasks (#2219)

KonradSzafer committed:
Add multiple chat template (#2129)

* multiple chat template support

* help doc update

* add transformers link to docstring

* model args update

* comment update

* statement simplification

* simplified chat_template property

* docs update

* removed template arg from HFLM class

* interface doc update

* model guide update

* interface doc update

* reuse apply_chat_template variable

* model guide refactor

* interface doc update

* removed old definition

* last nits

* last nits

* last nits

* better wording

* last nits

* Remove unnecessary Optional

* Apply suggestions from code review

Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

* return variable rename

---------

Co-authored-by: Clémentine Fourrier <22726840+clefourrier@users.noreply.github.com>
Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

haileyschoelkopf committed:
Update CODEOWNERS (#2229)

namtranase committed:
Fix Zeno Visualizer (#2227)

* fix: arguments data

* fix based on comment

* Update zeno_visualize.py

updated all output types

---------

Co-authored-by: Baber Abbasi <92168766+baberabb@users.noreply.github.com>

Geralt-Targaryen committed:
mela (#1970)

* mela

* Update mela_en.yaml

* Create _mela.yaml

---------

Co-authored-by: Lintang Sutawika <lintang@eleuther.ai>

lxning committed:
fix the regex string in mmlu_pro template (#2238)

* fix the regex string in yaml file

* Update samplers.py

---------

Co-authored-by: Lintang Sutawika <lintang@eleuther.ai>

WPoelman committed:
Fix logging when resizing embedding layer in peft mode (#2239)

baberabb committed:
computer_science --> "computer science" (#2241)

LSinev committed:
Fix typos in multiple places (#2244)

ACLUE bibtex typo reported to ACL Anthology and fixed here as title in pdf is correct.

eyuansu62 committed:
fix group args of mmlu and mmlu_pro (#2245)

Cameron7195 committed:
Created new task for testing Llama on Asdiv (#2236)

* Created DUP eval code for gsm8k

* asdiv

* Fixed fewshot=8 issue

* added results to .gitignore

* reverted unnecessary changes and moved results + gsm8k_dup out of repo to prepare for pull req

* fixed whitespace and unintentional hardcoded version change information

* created mbpp task

* Reverted changes re. mbpp to save for a future Pull req

* reverted metrics.py to previous commit

* updated asdiv readme to include informaiton about new asdiv_cot_llama task

* Apply suggestions from code review

---------

Co-authored-by: Alexander Detkov <alexander.d.detkov@gmail.com>
Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

baberabb committed:
chat template hotfix (#2250)

* chat template hotfix

* pre-commit

rimashahbazyan committed:
added Docker image
robustness same option eval
prompt robustness eval

rimashahbazyan committed:
added Docker image
robustness same option eval
prompt robustness eval

haileyschoelkopf committed:
[Draft] More descriptive `simple_evaluate()` LM TypeError (#2258)

* Update evaluator.py

* update error msg

haileyschoelkopf committed:
update nltk version to require 3.9.1 (#2259)

haileyschoelkopf committed:
Fix `loglikelihood_rolling` caching ( #1821 ) (#2187)

* fix revision type

* allow for None-input loglikelihood reqs to be cached

* handle no remaining cache items

* pre-commit

* change cache_hook.add_partial(loglikelihood_rolling...) convention

---------

Co-authored-by: Baber Abbasi <baber@eleuther.ai>

rimashahbazyan committed:
prompt loading updated

rimashahbazyan committed:
prompt loading updated

baberabb committed:
API: fix maxlen; vllm: prefix_token_id bug (#2262)

* max_length - 1 (generation always >= 1)

* vllm: fix rolling prefix_token

* nit: add comment

* fixup! max_length should be handled for logliklihoods

baberabb committed:
hotfix #2262 (#2264)

* max_length - 1 (generation always >= 1)

* vllm: fix rolling prefix_token

* nit: add comment

* fixup! max_length should be handled for logliklihoods

* Revert "fixup! max_length should be handled for logliklihoods"

This reverts commit 432d1a3b754c117c3a54ea2fe792ab3a1bd09ed3.

rimashahbazyan committed:
yaml based implementations of options and prompt robustness
+ consistency rate implementation

rimashahbazyan committed:
yaml based implementations of options and prompt robustness
+ consistency rate implementation

gnalbandyan commented on lm_eval/tasks/robustness/mmlu_pro/options_robustness_mmlu_pro.yaml:
```diff
@@ -0,0 +1,26 @@
+  max_gen_toks: 1024
+  do_sample: False
+  until:
+    - "<|eot_id|>"
```
this is model dependent, this is LLama's token, other models will have smth else

gnalbandyan commented on lm_eval/tasks/robustness/mmlu_pro/options_robustness_mmlu_pro.yaml:
```diff
@@ -0,0 +1,26 @@
+    - "<|eot_id|>"
+process_results: !function utils.options_robustness_process_results
+metric_list:
+  - metric: per_option_accuracy_std
```
can we have several? like mean max min std for example?

gnalbandyan commented on lm_eval/tasks/robustness/mmlu_pro/task.py:
```diff
@@ -0,0 +1,400 @@
+            Instance(
+                request_type="generate_until",
+                doc=doc,
+                arguments=(ctx, {"until": ["\n"], "max_gen_toks": 512}),
```
will the yaml override these?

gnalbandyan commented on lm_eval/tasks/robustness/mmlu_pro/task.py:
```diff
@@ -0,0 +1,400 @@
+        }
+
+
+class MmluProSameOption(ConfigurableTask):
```
is the difference between this and the task above only swapping labels? if yes, maybe use this as a child class, and all changes above in post processing, metrics etc will be applied here too

baberabb committed:
Chat Template fix (cont. #2235) (#2269)

* default chat template method fix

* move chat_template to TemplateLM

* remove hotfix

* handle openai `chat_template`

* Update lm_eval/api/model.py

Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

* add 'max_tokens' to gen_kwargs

* pre-commit

---------

Co-authored-by: KonradSzafer <szafer.konrad@gmail.com>
Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

gnalbandyan commented on lm_eval/tasks/robustness/mmlu_pro/utils.py:
```diff
@@ -0,0 +1,303 @@
+import re
+import numpy as np
+
+PROMPT_TEMPLATES_VERSION_02 = "v0.3_templates"
```
fix versions?

haileyschoelkopf committed:
Bump version to v0.4.4 ; Fixes to TMMLUplus (#2280)

rimashahbazyan committed:
fixing default generate_until (until) parameter to be eos of tokenizer

rimashahbazyan committed:
fixing default generate_until (until) parameter to be eos of tokenizer

rimashahbazyan committed:
minor refactorings

rimashahbazyan committed:
minor refactorings

rimashahbazyan committed:
added new templates

rimashahbazyan committed:
added new templates

rimashahbazyan committed:
v_03 templates

rimashahbazyan committed:
v_03 templates

rimashahbazyan committed:
macro accuracy for prompt robustness added

rimashahbazyan committed:
macro accuracy for prompt robustness added

rimashahbazyan committed:
option format robustness added

rimashahbazyan committed:
option format robustness added

Malikeh97 committed:
Add Open Arabic LLM Leaderboard Benchmarks (Full and Light Version) (#2232)

* arabic leaferboard yaml file is added

* arabic toxigen is implemented

* Dataset library is imported

* arabic sciq is added

* util file of arabic toxigen is updated

* arabic race is added

* arabic piqa is implemented

* arabic open qa is added

* arabic copa is implemented

* arabic boolq ia added

* arabic arc easy is added

* arabic arc challenge is added

* arabic exams benchmark is implemented

* arabic hellaswag is added

* arabic leaderboard yaml file metrics are updated

* arabic mmlu benchmarks are added

* arabic mmlu group yaml file is updated

* alghafa benchmarks are added

* acva benchmarks are added

* acva utils.py is updated

* light version of arabic leaderboard benchmarks are added

* bugs fixed

* bug fixed

* bug fixed

* bug fixed

* bug fixed

* bug fixed

* library import bug is fixed

* doc to target updated

* bash file is deleted

* results folder is deleted

* leaderboard groups are added

* full arabic leaderboard groups are added, plus some bug fixes to the light version

* Create README.md

README.md for arabic_leaderboard_complete

* Create README.md

README.md for arabic_leaderboard_light

* Delete lm_eval/tasks/arabic_leaderboard directory

* Update README.md

* Update README.md

adding the Arabic leaderboards to the library

* Update README.md

10% of the training set

* Update README.md

10% of the training set

* revert .gitignore to prev version

* Update lm_eval/tasks/README.md

Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

* updated main README.md

* Update lm_eval/tasks/README.md

* specify machine translated benchmarks (complete)

* specify machine translated benchmarks (light version)

* add alghafa to the related task names (complete and light)

* add 'acva' to the related task names (complete and light)

* add 'arabic_leaderboard' to all the groups (complete and light)

* all dataset - not a random sample

* added more accurate details to the readme file

* added mt_mmlu from okapi

* Update lm_eval/tasks/README.md

Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

* Update lm_eval/tasks/README.md

Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

* updated mt_mmlu readme

* renaming 'alghafa' full and light

* renaming 'arabic_mmlu' light and full

* renaming 'acva' full and light

* update readme and standardize dir/file names

* running pre-commit

---------

Co-authored-by: shahrzads <sayehban@ualberta.ca>
Co-authored-by: shahrzads <56282669+shahrzads@users.noreply.github.com>
Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

lintangsutawika committed:
Multimodal prototyping (#2243)

* add WIP hf vlm class

* add doc_to_image

* add mmmu tasks

* fix merge conflicts

* add lintang's changes to hf_vlms.py

* fix doc_to_image

* added yaml_path for config-loading

* revert

* add line to process str type v

* update

* modeling cleanup

* add aggregation for mmmu

* rewrite MMMU processing code based on only MMMU authors' repo (doc_to_image still WIP)

* implemented doc_to_image

* update doc_to_image to accept list of features

* update functions

* readd image processed

* update args process

* bugfix for repeated images fed to model

* push WIP loglikelihood code

* commit most recent code (generative ; qwen2-vl testing)

* preliminary image_token_id handling

* small mmmu update: some qs have >4 mcqa options

* push updated modeling code

* use processor.apply_chat_template

* add mathvista draft

* nit

* nit

* ensure no footguns in text<>multimodal LM<>task incompatibility

* add notification to readme regarding launch of prototype!

* fix compatibility check

* reorganize mmmu configs

* chat_template=None

* add interleave chat_template

* add condition

* add max_images; interleave=true

* nit

* testmini_mcq

* nit

* pass image string; convert img

* add vllm

* add init

* vlm add multi attr

* fixup

* pass max images to vllm model init

* nit

* encoding to device

* fix HFMultimodalLM.chat_template ?

* add mmmu readme

* remove erroneous prints

* use HFMultimodalLM.chat_template ; restore tasks/__init__.py

* add docstring for replace_placeholders in utils

* fix `replace_placeholders`; set image_string=None

* fix typo

* cleanup + fix merge conflicts

* update MMMU readme

* del mathvista

* add some sample scores

* Update README.md

* add log msg for image_string value

---------

Co-authored-by: haileyschoelkopf <hailey@eleuther.ai>
Co-authored-by: Baber Abbasi <baber@eleuther.ai>
Co-authored-by: Baber <baber@hey.com>
Co-authored-by: Hailey Schoelkopf <65563625+haileyschoelkopf@users.noreply.github.com>

rimashahbazyan committed:
Options order robustness fixed

rimashahbazyan committed:
Options order robustness fixed

rimashahbazyan committed:
Option format robustness fixed

rimashahbazyan committed:
Option format robustness fixed

rimashahbazyan committed:
metric names adjusted

rimashahbazyan committed:
metric names adjusted

rimashahbazyan committed:
per_option_macro_accuracy fix

rimashahbazyan committed:
per_option_macro_accuracy fix

SYusupov committed:
Update README.md (#2297)

* Update README.md

I encounter some Git buffer size limits when trying to download all commits history of the repository, such as:
```error: RPC failed; curl 18 transfer closed with outstanding read data remaining
error: 5815 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF```

therefore the installation is faster and there are not errors when I download only the last version of the repository

* Fix linting issue

baberabb committed:
repr bug (#2315)

dacorvo committed:
Update neuron backend (#2314)

* feat(neuron): align with latest optimum-neuron

* feat(neuron): support pre-exported neuron models

* fix(neuron): correctly use max_length

* fix(neuron): adapt loglikelihood

The evaluation of log likelihood was not working for neuron models
using continuous batching, such as all cached neuron LLama models.

* refactor(neuron): remove dead code

rimashahbazyan committed:
agi_eval added

rimashahbazyan committed:
agi_eval added

rimashahbazyan committed:
Filtering sampler added

rimashahbazyan committed:
Filtering sampler added

rimashahbazyan committed:
fewshot prompt robustness adedd and minor changes to other mmlu tasks

rimashahbazyan committed:
fewshot prompt robustness adedd and minor changes to other mmlu tasks

rimashahbazyan committed:
lsat_rc fix

rimashahbazyan committed:
lsat_rc fix

rimashahbazyan committed:
agieval_sat_math added

rimashahbazyan committed:
agieval_sat_math added

rimashahbazyan committed:
Filtering sampler refactored

rimashahbazyan committed:
Filtering sampler refactored

rimashahbazyan committed:
Minor refactornig and code dublicate removal
Added robustness_agieval and robustness_mmlu_pro groups

rimashahbazyan committed:
Minor refactornig and code dublicate removal
Added robustness_agieval and robustness_mmlu_pro groups

rimashahbazyan committed:
Readme added

rimashahbazyan committed:
Readme added

rimashahbazyan committed:
Merge branch 'robustness_task' of github.com:rimashahbazyan/lm-evaluation-harness into robustness_task
